app.py

```diff
@@ -242,10 +244,9 @@ def post(self, blobres):
     capdefault.put()
     pcapdata.put()
 
-    self.redirect('/d3viz.html?key=%s&to_plot=%s'
+    self.redirect('/d3viz.html#key~%s&to_plot~%s'
```

Comment:
So we definitely shouldn't do the ~ thing; you might have noticed that no other web app you've ever seen does this :)

However, it seems you're right and firefox acts differently from other browsers in this respect, and has since at least 2002(!): https://bugzilla.mozilla.org/show_bug.cgi?id=135309

I suggest using location.href instead of location.hash.  You'll have to by hand search for the # and then grab everything that came after the #, but at least it's the same in all browsers.